### PR TITLE
Addon - MFA

### LINKED ISSUE #86: Multi-Factor Authentication
- [x] Investigate best way to add multifactor-auth to the login ( google authenticator, SMS etc.)
- [x] Update config file as necessary.

### PR DESCRIPTION
**Pull request by jmike**
Fixes #86 

- Rename `CredentialsModel` to `AuthFactorModel`
- Use mongoose discriminators to separate between `PASSWORD` auth factor and `TOTP` auth factor
- Create API endpoints to manage TOTP auth factor
- Refactor session endpoints to use `secondaryFactorAuthentication` wherever necessary

### CODE REVIEW

File: server/api/totp/activate/controller.test.js [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+          secret,
+          token: '123456',
+        });
+
+      expect(res.status).toBe(200);
+      expect(res.body).toMatchObject({
+        ok: false,
+        error: {
+          code: 10027,
+          message: 'Secret key does not exist or has already expired',
```
Comment (spideynr) [inline server/api/totp/activate/controller.test.js]: This message is confusing me a little. If i try to activate a secret key, i get back that this secret does not exist?

File: server/api/totp/enroll/service.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  const secret = TOTPModel.generateSecret();
+
+  // update db
+  const session = await db.startSession();
+  session.startTransaction();
+
+  try {
+    // delete previous in-flight totp_enroll tokens (if any)
+    await TokenModel.deleteMany({
+      type: 'TOTP_ENROLL',
```
Comment (spideynr) [inline server/api/totp/enroll/service.js]: can we maybe eject these strings into a constant variable that we import wherever we need? Its fine now but i don't very much like magic

Comment (jmike) [reply server/api/totp/enroll/service.js]: Not magic, just fine-grained business-related logic. I am not sure how to make this clearer.

File: server/api/totp/activate/controller.test.js (hunk 1 above)
Comment (jmike) [reply server/api/totp/activate/controller.test.js]: Fixed.

File: server/api/totp/enroll/service.js (hunk 2 above)
Comment (jmike) [reply server/api/totp/enroll/service.js]: Fixed.

**Pull request merged**